pipeline/lib/pipeline.ml

```diff
@@ -121,11 +121,20 @@ let pipeline ~ocluster ~conninfo ~repository =
   and+ () =
     record_pipeline_stage ~stage:"run_job_id" ~serial_id ~conninfo worker
       worker_job_id
-  and+ output = Json_stream.save ~conninfo ~repository worker_job_id in
+  in
+  let output = Json_stream.save ~conninfo ~repository worker_job_id in
```

Comment:
I think you can define this `output` before the `let+`, and have a big chain of `and+` :

```ocaml
let output = Json_stream.save ~conninfo ~repository worker_job_id in
let+ () = record_pipeline_stage ... "build" ...
and+ () = record_pipeline_stage ... "run" ...
and+ () = match Current.state output with ... Error ... -> ... record_stage_failure ...
and+ out = output
in
out
```
This would allow to remove the star in `let* p = pipeline ...` below at line 137

(It absolutely works as is! But the less `let*` we have, the more predictable the pipeline is in the graphviz... ocaml-ci uses only `let+` and we are very close to having none too!)

Reply:
Thanks for the review and explanation, @art-w ! I've pushed an update changing this.